network-config-analyzer/ResourcesHandler.py

```diff
@@ -0,0 +1,233 @@
+    """
+    def __init__(self):
+        self.policies_finder = PoliciesFinder()
+        self.pods_finder = PodsFinder()
+        self.ns_finder = NamespacesFinder()
+        self.services_finder = ServicesFinder()
+
+    def parse_lists_for_topology(self, ns_list, pod_list, resource_list):
+        """
+        Chooses to parse the namespaces from the ns_list if exists in the input resources.
+        the pods and services: from pod_list if exist in the input resources
+        if the specified lists do not exist, try to parse from the resources list (if exists)
+        :param Union[list[str], None] ns_list: namespaces entry
+        :param Union[list[str], None] pod_list: pods entry
+        :param Union[list[str], None] resource_list: resources entry
+        :rtype (bool, ResourceType)
+        returns if succeeds or fails to find any topology objects as following:
+        True, 0 : if succeeds to find both namespaces and peers
+        False, ResourceType.Pods: if succeeds to find only pods
+        False, ResourceType.Namespaces: if succeeds to find only namespaces
+        False, 0: if fails to find both namespaces and pods
+        """
+        if ns_list:
+            ns_resource = ns_list
+            if resource_list:
+                print('Warning: namespaces will be taken from the provided namespace list key.'
+                      'input resources provided with resource list key will be ignored when finding namespaces')
+        else:
+            ns_resource = resource_list
+
+        if pod_list:
+            pod_resource = pod_list
+            if resource_list:
+                print('Warning: pods and services will be taken from the provided pod list key. '
+                      'input resources provided with resource list key will be ignored when finding pods and services')
+        else:
+            pod_resource = resource_list
+
+        if pod_resource is None and ns_resource is None:  # no resources to parse
+            return False, 0
+
+        if pod_resource == resource_list and pod_resource == ns_resource:  # both may exist in resourceList
+            self._parse_resources_path(resource_list,
+                                       [ResourceType.Namespaces, ResourceType.Pods])
+        else:  # we always want to parse for namespaces first (if exists)
+            if ns_resource:
+                self._parse_resources_path(ns_resource, [ResourceType.Namespaces])
+            if pod_resource:
+                self._parse_resources_path(pod_resource, [ResourceType.Pods])
+
+        if len(self.pods_finder.peer_set) > 0 and len(self.ns_finder.namespaces) > 0:
+            return True, 0
+        elif len(self.pods_finder.peer_set) == 0 and len(self.ns_finder.namespaces) > 0:
+            return False, ResourceType.Namespaces
+        elif len(self.pods_finder.peer_set) > 0 and len(self.ns_finder.namespaces) == 0:
+            return False, ResourceType.Pods
+        else:
+            return False, 0
```

Comment:
Why is it considered a failure if the length of pods or namespaces is 0?
Is the fallback in this case trying to get resources from live cluster?
Was this the behavior also in the previous version?
If the user specifies a path which as no pods resources at all, isn't it sufficient to report the size of the cluster topology (peer container)?

Reply:
before adding these "elif conditions" (from first commits), in case of "no namespaces/pods" are provided in the input resources, they'll be loaded from a k8s live cluster (this is also the behavior of old versions - always if ns_list/pod_list were not provided, then will be loaded from live cluster)

---------------------------------

these "elif" conditions were added **in case of "mixing global and private" resources in scheme files**.
for example, in scheme file, global resources may contain both pods and namespaces
but a specific network-config may override the ns_list --> in this case this method will find these private namespaces, and the fallback will use the global pods
in case no pods provided at all (neither global),  it will load them from k8s live cluster _( a fix is committed now regarding this)_

This keeps the origin behavior from old version for both cmdline and scheme tests

---------------------------------
so returning False in this method means "you can not build a peer container from the provided input resources" , with the resource type attached to the "False" result, the fallback knows to "fill" the missing parts

Reply:
If I try to run 'nca --sanity -r empty_file.yaml' then it would try to get resources from live cluster, as opposed to the behavior on master branch when running 'nca --sanity empty_file.yaml --ns_list empty_file.yaml --pod_list empty_file.yaml' , where the output would be:
global: cluster has 0 unique endpoints, 0 namespaces
global: cluster has 0 unique endpoints, 0 namespaces
No NetworkPolicies in empty.json. Nothing to check sanity on.

Reply:
the problem when providing a resource list , the "code" can not tell/predict what to search in it unless specific keys are provided.(specific keys that override it),

so if the resource list includes paths to namespaces and there are no pods at all (pod_list does not exist, and the resource_list does not include pods), it is hard to tell if the user intended to load pods from k8s or to run it with 0 pods.
 

I think I can tell the code that if "pod_list and ns_list" is provided "don't load from live cluster for empty files",
but what should I do when the objects are given with "resource_list"?

I would like to know your suggestion for these cases